File: backend/src/main/java/com/happy/friendogly/member/repository/MemberRepository.java
package com.happy.friendogly.member.repository;

import com.happy.friendogly.exception.FriendoglyException;
import com.happy.friendogly.member.domain.Member;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MemberRepository extends JpaRepository<Member, Long> {

    default Member getById(Long id) {
        return findById(id).orElseThrow(() -> new FriendoglyException("회원 정보를 찾지 못했습니다."));
    }
}

File: android/app/src/main/java/com/happy/friendogly/domain/repository/MemberRepository.kt
package com.happy.friendogly.domain.repository

import com.happy.friendogly.domain.DomainResult
import com.happy.friendogly.domain.error.DataError
import com.happy.friendogly.domain.model.Member
import com.happy.friendogly.domain.model.Register
import okhttp3.MultipartBody

interface MemberRepository {
    suspend fun postMember(
        name: String,
        accessToken: String,
        file: MultipartBody.Part?,
    ): DomainResult<Register, DataError.Network>

    suspend fun getMemberMine(): DomainResult<Member, DataError.Network>

    suspend fun getMember(id: Long): DomainResult<Member, DataError.Network>

    suspend fun deleteMember(): DomainResult<Unit, DataError.Network>
}
